src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -401,32 +404,32 @@ export class HttpConnection implements IConnection {
         }
     }
 
-    private resolveTransport(endpoint: IAvailableTransport, requestedTransport: HttpTransportType | undefined, requestedTransferFormat: TransferFormat): HttpTransportType | null {
+    private resolveTransport(endpoint: IAvailableTransport, requestedTransport: HttpTransportType | undefined, requestedTransferFormat: TransferFormat): HttpTransportType | Error {
```

Comment:
I see C/C++ has poisoned your mind to the point that you don't like throwing exceptions. I'm OK with this since this is fairly normal control flow.

I would like renaming this to tryResloveTransport, resolveTransportOrError or something like that. At least TS should prevent the caller from just assuming resolveTransport can only return a HttpTransportType.

Reply:
+1  to something like `resolveTransportOrError `
It would feel weird to have a "try"  prefixed method not return a bool.
Imo you should always be able to do 

```
if (tryFoo()) {
    // do something if Foo succeeded
}
else { 
    // do something if foo failed
}
```